crates/uniffi-bindgen-react-native/src/bindings/react_native/gen_typescript/templates/CallbackInterfaceImpl.ts

```diff
@@ -1,68 +1,72 @@
 {%- if self.include_once_check("CallbackInterfaceRuntime.ts") %}{%- include "CallbackInterfaceRuntime.ts" %}{%- endif %}
+{#
 {%- let trait_impl=format!("UniffiCallbackInterface{}", name) %}
 
 // Put the implementation in a struct so we don't pollute the top-level namespace
-fileprivate struct {{ trait_impl }} {
+class {{ trait_impl }} {
 
     // Create the VTable using a series of closures.
     // ts automatically converts these into C callback functions.
-    static var vtable: {{ vtable|ffi_type_name }} = {{ vtable|ffi_type_name }}(
+    static vtable: {{ vtable|ffi_type_name }} = new {{ vtable|ffi_type_name }}({
         {%- for (ffi_callback, meth) in vtable_methods %}
-        {{ meth.name()|fn_name }}: { (
+        {{ meth.name()|fn_name }}: (
             {%- for arg in ffi_callback.arguments() %}
             {{ arg.name()|var_name }}: {{ arg.type_().borrow()|ffi_type_name }}{% if !loop.last || ffi_callback.has_rust_call_status_arg() %},{% endif %}
             {%- endfor -%}
             {%- if ffi_callback.has_rust_call_status_arg() %}
             uniffiCallStatus: UnsafeMutablePointer<UniffiRustCallStatus>
             {%- endif %}
-        ) in
-            let makeCall = {
-                () {% if meth.is_async() %}async {% endif %}throws -> {% match meth.return_type() %}{% when Some(t) %}{{ t|type_name(ci) }}{% when None %}(){% endmatch %} in
-                guard let uniffiObj = try? {{ ffi_converter_name }}.handleMap.get(handle: uniffiHandle) else {
-                    throw UniffiInternalError.unexpectedStaleHandle
+        ) => {
+            const makeCall = {# space #}
+            {%- if meth.is_async() %}async {% else %} {% endif-%}
+            (): {% call ts::return_type(meth) %} => {
+                // Was converter.handleMap.get(uniffiHandle);
+                const uniffiObj = {{ ffi_converter_name }}.lift(uniffiHandle);
+                if (uniffiObj === undefined) {
+                    throw new UniffiInternalError.UnexpectedStaleHandle()
                 }
-                return {% if meth.throws() %}try {% endif %}{% if meth.is_async() %}await {% endif %}uniffiObj.{{ meth.name()|fn_name }}(
+                return {% if meth.is_async() %}await {% endif %}uniffiObj.{{ meth.name()|fn_name }}(
                     {%- for arg in meth.arguments() %}
-                    {{ arg.name()|arg_name }}: try {{ arg|lift_fn }}({{ arg.name()|var_name }}){% if !loop.last %},{% endif %}
+                    {{ arg|lift_fn }}({{ arg.name()|var_name }}){% if !loop.last %},{% endif %}
                     {%- endfor %}
                 )
             }
             {%- if !meth.is_async() %}
 
             {% match meth.return_type() %}
             {%- when Some(t) %}
-            let writeReturn = { uniffiOutReturn.pointee = {{ t|lower_fn }}($0) }
+            const writeReturn = (obj: any) => { uniffiOutReturn.pointee = {{ t|lower_fn }}(obj) };
```

Comment:
Is any a permanent thing here, or a temporary workaround -- not meant as a leading question, I understand that any-s are sometimes needed when building low-level bridges (and other systems are giving you typesafety here, so its not even really that bad).

Reply:
No, this who file is commented out. I'll definitely be revisiting this file.